src/index.js

```diff
@@ -27,25 +27,20 @@ var defaultOptions = {
  */
 SavitzkyGolay.prototype.calc = function (options) {
     options = options || {};
-    this.options = {};
-    for (var o in defaultOptions) {
-        if (options.hasOwnProperty(o)) {
-            this.options[o] = options[o];
-        } else {
-            this.options[o] = defaultOptions[o];
-        }
-    }
-    if ((this.options.windowSize % 2 === 0) || (this.options.windowSize < 5))
+    for (var o in defaultOptions)
+        if (!(options.hasOwnProperty(o)))
+            options[o] = defaultOptions[o];
+    if ((options.windowSize % 2 === 0) || (options.windowSize < 5))
         throw new RangeError('Invalid window size');
```

Comment:
Maybe you could be more explicit in the error (say that it has to be odd and at least 5)